core/src/genesis.rs

```diff
@@ -147,43 +147,46 @@ impl GenesisNetworkTrait for GenesisNetwork {
         genesis_config: Option<&Configuration>,
         tx_limits: &TransactionLimits,
     ) -> Result<Option<GenesisNetwork>> {
-        #![allow(clippy::unwrap_in_result)]
-        #![allow(clippy::expect_used)]
         if !submit_genesis {
             iroha_logger::debug!("Not submitting genesis");
             return Ok(None);
         }
         iroha_logger::debug!("Submitting genesis.");
-        Ok(Some(GenesisNetwork {
-            transactions: raw_block
-                .transactions
-                .iter()
-                .map(|raw_transaction| {
-                    let genesis_key_pair = KeyPair::new(
-                        genesis_config
-                            .as_ref()
-                            .expect("Should be `Some` when `submit_genesis` is true")
-                            .account_public_key
-                            .clone(),
-                        genesis_config
-                            .as_ref()
-                            .expect("Should be `Some` when `submit_genesis` is true")
-                            .account_private_key
-                            .clone()
-                            .ok_or_else(|| eyre!("Genesis account private key is empty."))?,
-                    )?;
-
-                    raw_transaction.sign_and_accept(genesis_key_pair, tx_limits)
-                })
-                .enumerate()
-                .filter_map(|(i, res)| {
-                    res.map_err(|error| {
-                        let error_msg = format!("{error:#}");
-                        iroha_logger::error!(error = %error_msg, "Genesis transaction #{i} failed")
```

Comment:
do we want to be forgiving in the genesis?

Reply:
I mean why don't we insist that genesis is completely valid?

Reply:
We only remove limitation on number of isi and wasm size in genesis block.

Genesis block is kinda special in a sense that it could be larger than ordinary block:
1) When @QuentinI will finish with his merging tool all block store will be condensed into one genensis block (and it probably be larger than normal block)
2) There is no time limit on applying genesis block (so we can afford it to larger imo)

Reply:
Let me rephrase. If a transaction in a genesis fails we seem to just skip it and log error. Why don't we fail blockchain initialization if there is even one invalid transaction?

Reply:
This is orthogonal question to this PR. I understand that you've relaxed limits on genesis

Reply:
Oh, i see now, this is actually good point. IMO we shouldn't allow such behavior (what point in doing this after all).

Reply:
Let's open a ticket to discuss further. There may be a good reason for this

Reply:
Created #3075.

Reply:
Actually after relaxation on limits only one reason for invalid tx would be wrong signatures.